internal/xds/translator/translator.go

```diff
@@ -772,6 +772,10 @@ func processXdsCluster(tCtx *types.ResourceVersionTable, httpRoute *ir.HTTPRoute
 		clusterArgs.backendConnection = bt.BackendConnection
 	}
 
+	if httpRoute.DNS != nil {
```

Comment:
is something similar needed for grpc, tcp, udp, tls ?

Reply:
Its probably can be still applicable If the route specifies a DNS hostname, it should function for all protocols within the context of the Envoy gateway cluster.

Reply:
@arkodg  - do you want to create an issue following DNS setting implementation across routes and non-routes? I think that we can start out with limited support for HTTP routes.

Reply:
yeah for xRoutes is fine for now, what I was trying to get at this, the code is not uniform, for e.g. for TCP Route translation we have another code flow https://github.com/envoyproxy/gateway/blob/cbabdc623acb1c60c9e3790e9e9d870f117eec95/internal/xds/translator/translator.go#L576 so we missed adding it there

Reply:
right, thanks for catching it. I have just added that for tcp and udp routes.

Reply:
thanks @alexwo can you help with the tech debt here , is it possible to reuse `processXdsCluster` for other route types, so we dont miss something like this again

Reply:
Sure, I have modified processXdsCluster to be compatible with any type that can return cluster arguments,  currently used for TCP, UDP, and HTTP routes.​⬤